packages/grafana-ui/src/components/Graph/Graph.tsx

```diff
@@ -101,15 +101,18 @@ export class Graph extends PureComponent<GraphProps, GraphState> {
     if (series.length === 0) {
       return [{ show: true, min: -1, max: 1 }];
     }
-    return uniqBy(
+
+    const showLeftYAxis = series.some(s => s.yAxis.index === 1);
+
+    return _.uniqBy(
```

Comment:
Logic here was changed so that if we have both logs and metrics results in a graph, the logs Y-Axis will show on the left and the metrics Y-Axis will show on the right, but if there's no logs data the metrics Y-Axis will show on the left. Might be a bit difficult to parse though so I'm considering re-writing this to make it more clear

Reply:
What is the usecase of having "logs in a graph"? I mean for me logs counts would be just another metric so I would not think this should be treated separately. At the same time though @davkal merged this PR https://github.com/grafana/grafana/pull/25027 which I think may change the reasoning for logs counts.

Imho I would rather have logs counts as part of logs component, and just count the logs shown (which would mean we would remove a special query in elastic and just compute the counts front end same as loki). This would mean logs counts would be simplified because it would not concern datasource at all it would be just part of the visualisation computed over the same data.

This would free us to treat any other metrics response as simple metrics and we would not need to check if the metrics is logs count or any other metric.